packages/web-app-files/src/helpers/resource/resource.ts

```diff
@@ -57,3 +58,24 @@ export const extractStorageId = (id?: string): string => {
   }
   return id.indexOf('!') >= 0 ? id.split('!')[0] : ''
 }
+
+export const extractNameWithoutExtension = (resource?: Resource): string => {
+  const extension = resource.extension || ''
+  const name = resource.name || ''
+  if (!extension.length || !name.length) return name
+  const extensionIndexInName = name.lastIndexOf(`.${extension}`)
+  return name.substring(0, extensionIndexInName)
+}
+
+export const extractExtensionFromFile = (file: string): string => {
```

Comment:
Since this is in the `resource.ts` helper, I'd expect that you don't work with a name string, but with a `Resource` here, like in the other function.
First step: check if `resource.type === 'folder'`, if yes return empty string.
Second step: split the file name on a `.`, if it only has two parts you can immediately return the second part. With those two steps we return super quick 99% of the time.
Only after that it makes sense to go into the complex extension checks.